intro-js.md

```diff
@@ -0,0 +1,222 @@
+
+An issue with ECMAScript is that not all browsers are fully capable of supporting the latest versions of ECMAScript. 
+
+This creates a problem because it means that native support is not available for the features within the browser itself.
+
+There is a workaround to this that will be introduced to you as the lessons move forward. 
+
+For now, take a look at this [website](http://caniuse.com/#search=es6) to see what browsers support the latest versions of ECMAScript. 
+
+You don't need to understand that web page completely, but you should have one takeaway from it. 
+
+Stay away from Internet Explorer!
+
+What does ECMAScript give us? 
+
+ECMAScript defines standards for features that are included in JavaScript. 
+
+For example, within earlier versions of JavaScript, classes were not defined with the ```class``` keyword. 
+
+There was a slightly more complicated route to defining classes in JavaScript. However, with ES6 onwards you can now define a class using the ```class``` keyword. 
+
+This does not mean that classes work the same way in JavaScript as they do in other OOP languages. 
+
+It simply means that the native way of defining classes in JavaScript has been extended to include a way to define classes with the ```class``` keyword. 
+
+That is syntactic sugar. 
+
+So what exactly is syntactic sugar? 
+
+Syntactic sugar is basically helpful syntax to make developer's lives easier for development, but in the end, in terms of functionality differences, there is none. 
+
+So, getting back to classes. 
+
+Example class
+```javascript 
+class Shape {
+    constructor(param1, param2) {
+        
+    }
+}
+```
+
+The constructor method is what is used to create an instance of the class Shape above. 
+
+You can only have one constructor method per class. 
+
+The parameters for the constructor [param1, param2] are also optional. You could just as easily define the class like below. 
+
+```javascript
+class Shape {
+    constructor() {
+        
+    }
+}
+```
+
+The params represent the properties you want to set on your object. 
+
+### Functions are first class objects
+
+This one is a little hard to wrap your heads around so we won't go into too much detail, but it is one of the most fundamental aspects of JavaScript. 
+
+It is what makes JavaScript such a powerful language. 
+
+Basically, what this means is that functions can be constructed during execution, they can be passed around as variables or parameters to other functions and you can also return a function as the value of another function (what is referred to as currying). 
+
+Confused? That's okay. If you wish to, you can read more about it [here](http://timmknight.github.io/2015/first-class-functions-javascript/)
+
+There are a lot of other reasons, but these will be introduced as the meetings move forward. 
+
+## Basic data types in JavaScript
+Some of the ECMAScript standard datatypes:
+* Boolean 
+* Number
+* String
+
+# JSON
```

Comment:
I think that's okay. Let's get some of these ideas written up so we can see what it looks like...this is ending up as a megathread